File: bothunder/Evynd_Firestorm.pl
# NPCID: Bastion of Thunder >> Evynd_Firestorm (209054)
sub EVENT_AGGRO {
	# tell the A_firestorm_portal I have engaged
	quest::signalwith(209038,1,1);
	# create a timer to check if I have lost all aggro and need to reset the event
	quest::settimer("evynd_aggro_check",60);
}

sub EVENT_DEATH_COMPLETE {
	# spawn #Askr_the_Lost to port players to the next floor
	quest::spawn2(209101,0,0,$x,$y,$z,$h);
	# tell the A_firestorm_portal I have died
	quest::signalwith(209038,2,1);
}

sub EVENT_TIMER {
	if ($timer eq "evynd_aggro_check") {
		if(!$npc->IsEngaged()) {
			# if I have no aggro stop the portals from spawning adds
			quest::signalwith(209038,2,1);
			#depop the firestorm portal adds.
			quest::depopall(209118);
			quest::stoptimer("evynd_aggro_check");
		}
	}
}
		

File: bothunder/Evynd_Firestorm.lua
function event_combat(e)
	if (e.joined == true) then
		eq.set_timer("Evynd_Adds", 120 * 1000);
	else
		eq.stop_timer("Evynd_Adds");
	end
end

function event_timer(e)
	if (e.timer == "Evynd_Adds") then
		eq.signal(209038 ,1,1); -- NPC: A_firestorm_portal
		eq.zone_emote(0, "Evynd waves his hands in the air, calling the power of the firestorm through the blazing portals.");
	end
end

function event_death_complete(e)
	eq.spawn2(209101,0,0,e.self:GetX(),e.self:GetY(),e.self:GetZ(),e.self:GetHeading()); -- NPC: #Askr_the_Lost
end
